Source/Scene/Expression.js

```diff
@@ -56,8 +56,11 @@ define([
 
     });
 
-    Expression.prototype.evaluate = function(feature) {
-        return this._runtimeAst.evaluate(feature);
+    Expression.prototype.evaluate = function(feature, result) {
```

Comment:
It would be called from here: https://github.com/AnalyticalGraphicsInc/cesium/blob/3d-tiles-style/Source/Scene/Cesium3DTileStyleEngine.js#L119

Individual nodes may also need scratch variables for the case you mentioned.